pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -0,0 +1,253 @@
+	arr = append(arr, attachedDisk)
+
+	network := &compute.NetworkInterface{
+		Network:       "projects/" + opt.ProjectID + "/global/networks/" + opt.Shoot.Name,
+		Subnetwork:    "regions/" + opt.Shoot.Spec.Region + "/subnetworks/" + opt.SubnetWork,
+		AccessConfigs: []*compute.AccessConfig{{Name: "External NAT", Type: "ONE_TO_ONE_NAT"}},
+	}
+	networkArr := make([]*compute.NetworkInterface, 1)
+	networkArr = append(networkArr, network)
+	machineType := fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/zones/%s/machineTypes/n1-standard-1", opt.ProjectID, opt.Zone)
+
+	insta, err := gcpclient.Instances().Get(opt.ProjectID, opt.Zone, opt.BastionInstanceName).Context(ctx).Do()
+	if err != nil {
+		gerr := err.(*googleapi.Error)
+		if gerr.Code != 404 {
+			return nil, errors.Wrap(err, "failed to get compute instance")
+		}
+	}
+	metadataItems := make([]*compute.MetadataItems, 0)
+	metadataItems = append(metadataItems, &compute.MetadataItems{
+		Key:   "startup-script",
+		Value: &opt.UserData,
+	})
+
+	if insta != nil {
+		logger.Info("existing bastion compute instance found", "compute_instance_name", insta.Name)
+	} else {
+		instance := &compute.Instance{
+			Disks:              arr,
+			DeletionProtection: false,
+			Description:        "gardenctl bastion Instance",
+			Name:               opt.BastionInstanceName,
+			Zone:               opt.Zone,
+			MachineType:        machineType,
+			NetworkInterfaces:  networkArr,
+			Tags:               &compute.Tags{Items: []string{"gardenctl"}},
+			Metadata:           &compute.Metadata{Items: metadataItems},
+		}
+
+		_, err = gcpclient.Instances().Insert(opt.ProjectID, opt.Zone, instance).Context(ctx).Do()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create instance")
+		}
+	}
+
+	return getInstanceEndpoints(ctx, gcpclient, opt)
+}
+
+func getInstanceEndpoints(ctx context.Context, gcpclient gcpclient.Interface, opt *Options) (*bastionEndpoints, error) {
+	instance, err := getBastionInstance(ctx, gcpclient, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	if instance == nil {
+		return nil, nil
+	}
+
+	endpoints := &bastionEndpoints{}
+
+	if ingress := addressToIngress(&instance.Name, &instance.NetworkInterfaces[0].NetworkIP); ingress != nil {
+		endpoints.private = ingress
+	}
+
+	if instance.NetworkInterfaces == nil || len(instance.NetworkInterfaces) == 0 {
+		err := errors.New("no network interfaces found")
+		logger.Error(err, "no network interfaces found", "compute_instance", instance.Name)
+		return nil, err
+	}
+
+	if instance.NetworkInterfaces[0].AccessConfigs == nil || len(instance.NetworkInterfaces[0].AccessConfigs) == 0 {
+		err := errors.New("no access config found for network interface")
+		logger.Error(err, "no access config found for network interface", "compute_instance", instance.Name)
+		return nil, err
+	}
+
+	if ingress := addressToIngress(&instance.Name, &instance.NetworkInterfaces[0].AccessConfigs[0].NatIP); ingress != nil {
```

Comment:
oh I see. on my local I only use a bastion Public IP address to test the whole ssh flow.  I will take a look at this to make host domainName working as well.

Reply:
I had an experiment today refer to https://cloud.google.com/compute/docs/instances/create-ptr-record#api, then add the following code during VM creation. one error message during the creation before VM created `Error 400: Invalid value for field 'resource.networkInterfaces[0].accessConfigs[0].publicPtrDomainName': 'default-xxxxx-xxxxx-bastion.c.sxx-sx-gcp-scp-k8s-dev.internal'. Please verify ownership of the PTR domain at https://www.google.com/webmasters/verification/verification?domain=default-xxxxx-xxxxx-bastion.c.sxx-sx-gcp-scp-k8s-dev.internal, invalid, failed to create instance,`  same error when I create this from the vm instance portal page manually. 

I am feeling to make the Public PTR domain work, We may need to have a verified domain name first eg `default-xxxxx-xxxxx-bastion.c.sxx-sx-gcp-scp-k8s-dev.internal` ,  from the cloud domain management page, I observed this feature not enabled in project  https://console.cloud.google.com/net-services/domains/registrations/list?project=sxx-sx-gcp-scp-k8s-dev.  that mean we need to enable this service? create/manage a verified domain every time when new bastion VM is created.  Personally, I think less resources are used, the lower risk for us 

yes, the `instance.Name` not work and Public PTR domain seems not able to use as a public hostname simply, Should I make it empty like this here to avoid misleading? Any comments about this 

```
NAME            IP            HOSTNAME   AGE
xxxx-bstion   34.77.212.7              41s
```

```
Name:                "External NAT",
Type:                "ONE_TO_ONE_NAT",
SetPublicPtr:        true,
PublicPtrDomainName: opt.BastionInstanceName + ".c." + opt.ProjectID + ".internal"
```

Reply:
ok then lets leave the public dns name nil for now and tackle it in the future in case we really need it

Reply:
committed in c1472d2ede0f5d6334526fa6fb08c4e719c43b4b